### PR TITLE
MongoDB vs Firebase

### PR DESCRIPTION
**Pull request by kenny98dev**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋  Good evening @kenny98dev  and welcome to the EngEd program. 
Happy to have you join us. Thank you for contributing your 1st article. 
Upon running your PR through our 3rd party plagiarism checker it raised a few red flags (see attached) & the % was a bit higher than we typically accept.
Please revisit the article and change any underlined sentences and/or and sources where needed 🚀  
As we need to ensure we are contributing original content to the community and to our audience.
Thank you
[MongoDB vs Firebase #516.pdf](https://github.com/section-io/engineering-education/files/5442577/MongoDB.vs.Firebase.516.pdf)

**Comment by hectorkambow** [comment]
Author pinged via Slack - NOTE

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/choosing-between-firebase-and-mongo-db/hero.jpg
+    alt: Relational Databases
+---
+
+### MongoDB vs Firebase
+Databases play an important role in determining the functionality and quality of an application or a website. For an app to be successful,
+the developer will have to consider the users of the application or the website.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:19]: Spacing irregularity

Comment (sudosoph) [reply articles/choosing-between-firebase-and-mongodb/index.md:19]: "The developer will have to consider the users of the application of the website" => very generic. Replace instead with why databases are crucial to app development to give some context.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+The quantity of data stored and retrieved from the application is one of the factors considered when choosing a database.
+Other factors considered when choosing the right database for an application or a website are:
+1. Latency.
+Latency refers to the response time of the database and to the end-to-end response time of the application. 
+2.structure of the data. Choosing the right database for an application or a website involves the developer understanding the right data structures for storage and retrieval of data. Various applications deal with data of various formats In this article, I am going to focus on Firebase and MongoDB while highlighting its differences.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:24]: Check capitalization.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  - url: /engineering-education/choosing-between-firebase-and-mongo-db/hero.jpg
+    alt: Relational Databases
+---
+
+### MongoDB vs Firebase
+Databases play an important role in determining the functionality and quality of an application or a website. For an app to be successful,
+the developer will have to consider the users of the application or the website.
+The quantity of data stored and retrieved from the application is one of the factors considered when choosing a database.
+Other factors considered when choosing the right database for an application or a website are:
+1. Latency.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:22]: Formatting. Use a colon and bold the word when writing a category or definition. 

i.e.

1. **Latency**: Response time of the database...

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+Latency refers to the response time of the database and to the end-to-end response time of the application. 
+2.structure of the data. Choosing the right database for an application or a website involves the developer understanding the right data structures for storage and retrieval of data. Various applications deal with data of various formats In this article, I am going to focus on Firebase and MongoDB while highlighting its differences.
+### History of MongoDB
+10gen company which is now MongoDB Inc developed MongoDB in the year 2007. The 10gen main language was server-side JavaScript.10gen company started to develop as a Platform as a service ( PAAS) product, but they experienced scalability issues with the existing relational database systems. They developed a document-oriented database system called MongoDB.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:26]: This doesn't make sense. You are trying to say that 10gen was the original company that developed MongoDB in 2007 and changed their name.

"10gen company" is not English. => "The company 10gen"

Capitalize Platform as a Service
Spacing around (PAAS)

What were the existing relational database systems?

What does document-oriented mean?

File: articles/choosing-between-firebase-and-mongodb/index.md (hunk 3 above)
Comment (sudosoph) [reply articles/choosing-between-firebase-and-mongodb/index.md:22]: What are other factors in choosing a database? Overview.

File: articles/choosing-between-firebase-and-mongodb/index.md (hunk 2 above)
Comment (sudosoph) [reply articles/choosing-between-firebase-and-mongodb/index.md:24]: ...and to the end-to-end response time of the application. => doesn't make sense

"an application or a website" -> redundant, pick one and be consistent

various formats In this article => run-on

"...while highlighting its differences" -> you are talking about 2 things, should be plural

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+4. MongoDB's documentation has an extensive collection of literature and MongoDB tutorials for new users.
+5. MongoDB has high data security since there is no risk of SQL injection.
+### Disadvantages of MongoDB
+1. There is high data consumption because of denormalization.
+2. It has no joins. We need to make many queries and join data.
+3. MongoDB has a limited data size.
+4. Indexing and searching in MongoDB is not powerful.
+### History of Firebase
+Firebase is a cloud service provider and backend as a service company based in San Francisco, California, founded in September 2011 by James Tamplin and Andrew Lee as Envolve. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website.
+### What is firebase
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:53]: Capitalization for headers, punctuation

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+### Advantages of Firebase
+1. It is free.
+Developers can create as more than one project in Firebase for free. Developers don't need to buy a premium license to use Firebase.
+2. Concise Documentation
+Firebase SDK is documented and has plenty of examples over its website.
+3. Improved Database
+Firebase is an improved database since it provides better performance by storing large chunks of data in JSON format.
+4. It is fast.
+Applications and websites developed using Firebase are faster because its cloud services lead to faster fetching and displaying of data on the screens.
+### Disadvantages of using Firebase
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:72]: Above you wrote "Advantages of Firebase"

Make this header parallel by removing "using"

File: articles/choosing-between-firebase-and-mongodb/index.md (hunk 4 above)
Comment (sudosoph) [reply articles/choosing-between-firebase-and-mongodb/index.md:26]: Why and how did they develop MongoDB? You can go into the origin story and cite your sources.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+1. Latency.
+Latency refers to the response time of the database and to the end-to-end response time of the application. 
+2.structure of the data. Choosing the right database for an application or a website involves the developer understanding the right data structures for storage and retrieval of data. Various applications deal with data of various formats In this article, I am going to focus on Firebase and MongoDB while highlighting its differences.
+### History of MongoDB
+10gen company which is now MongoDB Inc developed MongoDB in the year 2007. The 10gen main language was server-side JavaScript.10gen company started to develop as a Platform as a service ( PAAS) product, but they experienced scalability issues with the existing relational database systems. They developed a document-oriented database system called MongoDB.
+### Introduction to MongoDB
+MongoDB is an open-source document-oriented database that stores data in JSON-like formats. MongoDB makes use of documents and collections instead of using tables and rows. It supports the dynamic schema design. MongoDB is a document-oriented server developed in a C++ programming language.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:28]: Briefly explain what a JSON-like format entails so the reader knows why it was chosen vs. XML or other formats.

Define documents and collections.

What is dynamic schema design?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+1. Native Replication.
+MongoDB provides a replication feature by distributing data across different machines through a replica set topology.According to [Congruent Solutions](https://www.congruentsolutions.com/blogposts/mongodb-replication-approach-setup-using-arbiter/),A replica set is a group of MongoDB instances that host the same data set.In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries. Replica set can have only one primary node.After each successive writes to the primary node, MongoDB will replicate the same data to all the secondary nodes present in the replica set.
+2. Supports Ad-hoc Queries.
+MongoDB supports ad-hoc queries by indexing the Binary JSON ( BSON) documents.
+3. It is schemaless.
+MongoDB is considered flexible since its database is written in the C++ programming language.
+4. High Performance.
+MongoDB's performance is high as compared to other databases due to its scalability, and it has a faster query response because of replication and indexing thus making it a better choice for real-time applications.
+5. Indexing.
+MongoDB indexes improve search performance.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:39]: How??

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+### Introduction to MongoDB
+MongoDB is an open-source document-oriented database that stores data in JSON-like formats. MongoDB makes use of documents and collections instead of using tables and rows. It supports the dynamic schema design. MongoDB is a document-oriented server developed in a C++ programming language.
+### Features of MongoDB
+1. Native Replication.
+MongoDB provides a replication feature by distributing data across different machines through a replica set topology.According to [Congruent Solutions](https://www.congruentsolutions.com/blogposts/mongodb-replication-approach-setup-using-arbiter/),A replica set is a group of MongoDB instances that host the same data set.In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries. Replica set can have only one primary node.After each successive writes to the primary node, MongoDB will replicate the same data to all the secondary nodes present in the replica set.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:31]: Replica set topology - break down

Spacing -> "Accordingly..."

Odd capitalization + spacing -> ",A replica"

Spacing throughout??? There are no spaces after periods.

In a replica set one node-> run on, needs comma

"In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries." => what does this mean??

"Replica set can have only one primary node." -> GRAMMAR. 

The concept of why there is only one primary node is confusing - simplify.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+2. Supports Ad-hoc Queries.
+MongoDB supports ad-hoc queries by indexing the Binary JSON ( BSON) documents.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:33]: Spacing.

What are ad-hoc queries?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+3. It is schemaless.
+MongoDB is considered flexible since its database is written in the C++ programming language.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:35]: What does schemaless mean? What does this have to do with flexibility?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+4. High Performance.
+MongoDB's performance is high as compared to other databases due to its scalability, and it has a faster query response because of replication and indexing thus making it a better choice for real-time applications.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:37]: How do you define high performance? Metrics, statistics??

"and indexing thus making it a..." -> missing comma

Better choice as opposed to what?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+5. Indexing.
+MongoDB indexes improve search performance.
+### Advantages of MongoDB
+1. It contains heterogeneous data.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:41]: What is heterogeneous data? Of what type(s)?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+2. It provides high performance, scalability, and availability.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:42]: You mention performance + scalability briefly above but never availability. What does that mean?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+3. It can be integrated with Big Data Hadoop.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:43]: Commonly referred to as just Hadoop (not Big Data Hadoop) - what is Hadoop and why should the reader care?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+4. MongoDB's documentation has an extensive collection of literature and MongoDB tutorials for new users.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:44]: Wordy. "MongoDB has extensive documentation and tutorials for new users"

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+5. MongoDB has high data security since there is no risk of SQL injection.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:45]: "high data security" = awkward phrasing. Rephrase.

Why is there no risk of SQL injection? Elaborate...

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+### Disadvantages of MongoDB
+1. There is high data consumption because of denormalization.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:47]: What is denormalization?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+2. It has no joins. We need to make many queries and join data.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:48]: How does one manually join data and why is this a disadvantage, compared to, say, SQL? How is SQL's join command implemented instead? (See $lookup command)

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+3. MongoDB has a limited data size.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:49]: What is the size?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+4. Indexing and searching in MongoDB is not powerful.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:50]: Why not?

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+### History of Firebase
+Firebase is a cloud service provider and backend as a service company based in San Francisco, California, founded in September 2011 by James Tamplin and Andrew Lee as Envolve. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:52]: Spacing.

You don't have to write out "Application Programming Interface(API)" multiple times. You can write it once and use API from there on. Check spacing.

You repeat "enabled the integration of online chat functionality into their website." twice - delete

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+|MongoDB is a Platform-as-a-Service solution.| Firebase is a Backend-as-a-Service solution.|
+|MongoDB is suitable for large-scale applications.| Firebase is suitable for small-scale applications.|
+|MongoDB is an open-source database. | Firebase is an open-source database.|
+|Proprietary protocol using JSON are used as APIs and other access methods.|Android, iOS, JavaScript API, RESTful HTTP API are used as APIs and other access methods.|
+|Server operating systems for MongoDB are Solaris, Linux, OS X, Windows. | Server operating systems for Firebase are hosted.|
+### Conclusion
+Every database has its unique features, and it is designed to provide solutions for different problems. The choice of each database depends on its benefits and drawbacks, and also the business requirements. Both MongoDB and Firebase have their pros and cons. After all the advantages, and disadvantages, comparisons and the description, the developer will have to choose which database they can integrate into their application or website.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:95]: Generic. This should be a recap of the specific similarities and differences you discuss, as a summary.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+2. Costs
+Firebase free plan limits up to 50 connections and 1,000 Megabytes of storage.
+3. Data Migration Problem.
+Firebase stores its data in JSON format so managing large amounts of structured data need the developer to write powerful queries in SQL.
+4. Inconvenient Data Storage
+Firebase cannot handle complex query structures. It limits the developer to combine query options. Because of this limitation, one needs to arrange the documents by date or filter them using a search query.
+5. No staging environment.
+Firebase does not allow the developer to test the database in a staging environment.
+### Differences betweeen MongoDB and Firebase
+According to [GeeksforGeeks](https://www.geeksforgeeks.org/difference-between-firebase-and-mongodb/), the main differences between MongoDB  and Firebase are:
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:84]: Spacing

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+|MongoDB|Firebase|
+|---|---|
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:86]: Table formatting spacing

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+|MongoDB Inc developed MongoDB.| Google developed Firebase|
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:87]: Parallelism

Developed by MongoDB/Developed by Google
Supports C, C#.../Supports Objective-C, Java...

etc.

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 27]
```diff
@@ -0,0 +1,95 @@
...
+5. MongoDB has high data security since there is no risk of SQL injection.
+### Disadvantages of MongoDB
+1. There is high data consumption because of denormalization.
+2. It has no joins. We need to make many queries and join data.
+3. MongoDB has a limited data size.
+4. Indexing and searching in MongoDB is not powerful.
+### History of Firebase
+Firebase is a cloud service provider and backend as a service company based in San Francisco, California, founded in September 2011 by James Tamplin and Andrew Lee as Envolve. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website. It provided developers with an Application Programming Interface(API) that enabled the integration of online chat functionality into their website.
+### What is firebase
+According to [Cypress](https://www.cypressoft.com/post/choose-firebase-over-backend-development/),Firebase is a toolkit and infrastructure that aims at supporting the process of building better applications and websites.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:54]: Spacing

See above comment on "applications and websites"

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+Firebase is a Back-end as a service owned by Google which provides a server-less backend to app and web developers. Firebase offers a wide range of backend functionalities which include authentication,real-time database, storage, analytics, crash reporting, and dynamic links.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:55]: First sentence -> Already mentioned above

Keep second sentence.

Unclear whether those features are also in MongoDB or not

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 29]
```diff
@@ -0,0 +1,95 @@
...
+Firebase is a hosted solution so the user doesn't have any root access to the location where the data is stored.
+2. Costs
+Firebase free plan limits up to 50 connections and 1,000 Megabytes of storage.
+3. Data Migration Problem.
+Firebase stores its data in JSON format so managing large amounts of structured data need the developer to write powerful queries in SQL.
+4. Inconvenient Data Storage
+Firebase cannot handle complex query structures. It limits the developer to combine query options. Because of this limitation, one needs to arrange the documents by date or filter them using a search query.
+5. No staging environment.
+Firebase does not allow the developer to test the database in a staging environment.
+### Differences betweeen MongoDB and Firebase
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:83]: Typo

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 30]
```diff
@@ -0,0 +1,95 @@
...
+1. It is free.
+Developers can create as more than one project in Firebase for free. Developers don't need to buy a premium license to use Firebase.
+2. Concise Documentation
+Firebase SDK is documented and has plenty of examples over its website.
+3. Improved Database
+Firebase is an improved database since it provides better performance by storing large chunks of data in JSON format.
+4. It is fast.
+Applications and websites developed using Firebase are faster because its cloud services lead to faster fetching and displaying of data on the screens.
+### Disadvantages of using Firebase
+1. Hosted solution
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:73]: Capitalization

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 31]
```diff
@@ -0,0 +1,95 @@
...
+Firebase is a hosted solution so the user doesn't have any root access to the location where the data is stored.
+2. Costs
+Firebase free plan limits up to 50 connections and 1,000 Megabytes of storage.
+3. Data Migration Problem.
+Firebase stores its data in JSON format so managing large amounts of structured data need the developer to write powerful queries in SQL.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:78]: JSON format -> just "JSON"

Need commas before clauses i.e. "so"

"need the developer" => replace need with require

File: articles/choosing-between-firebase-and-mongodb/index.md [hunk 32]
```diff
@@ -0,0 +1,95 @@
...
+4. Inconvenient Data Storage
+Firebase cannot handle complex query structures. It limits the developer to combine query options. Because of this limitation, one needs to arrange the documents by date or filter them using a search query.
+5. No staging environment.
+Firebase does not allow the developer to test the database in a staging environment.
```
Comment (sudosoph) [inline articles/choosing-between-firebase-and-mongodb/index.md:82]: How do developers test?

**Review by sudosoph — CHANGES_REQUESTED** [review]
- Several spacing, capitalization, and grammatical issues.
- Missing a lot of definitions and elaborations.
- Only one source mentioned.
- Parallelism in headers and table formats require work.
- Assume the reader is a beginner and make sure to define and explain things. Needs to be more informational.

**Comment by hectorkambow** [comment]
@kenny98dev just pinging you to make sure you saw our Peer reviewer notes/suggestions on how to improve the article.

**Comment by hectorkambow** [comment]
Just closing PR to make room in the open PR queue. This PR can be re-opened at anytime once ready 🚀 
Thank you

**Pull request closed without merging**